packages/wethegit-react-hooks/src/lib/hooks/use-in-view.ts

```diff
@@ -50,20 +62,22 @@ export function useInView<T extends HTMLElement>(
   )
 
   useEffect(() => {
-    if (observerRef.current) return
-
-    if (!targetRef) return
-
-    const optionsRef = typeof threshold === "number" ? { threshold } : threshold
-
-    observerRef.current = new IntersectionObserver(observerCallback, optionsRef)
+    if (observerRef.current || !targetRef) return
 
+    observerRef.current = new IntersectionObserver(observerCallback, settings)
     observerRef.current.observe(targetRef)
 
     return () => {
       if (observerRef.current) observerRef.current.unobserve(targetRef)
     }
-  }, [observerCallback, targetRef, threshold])
+
+    // Important: `settings` must be kept out of this dependency Array.
+    // If that variable changes after the observer is already hooked up
+    // (strict mode is a good use case for when this would occur), then the
+    // observer will catch the early return and have already disconnected.
```

Comment:
Just to confirm, you are saying that `targetRef` will change anyways if `settings` change and the component re-renders and so we don't need for it to be part of the dependency array?

Reply:
My comment there was because of the render cycle actually. In our useEffect, we clean up after the render phase, disconnecting the observer. Because of that, if `settings` changes — causing the effect to run again — the early return will happen, since we have `if (observerRef.current) return`. This basically means that the observer will remain disconnected/unobserved, despite the settings having changed.

Maybe there's a better solution here though? Like, can we check the `connectedness` of the observer, instead of the object itself.

Reply:
Check [this part](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/disconnect) of the API. Maybe the observer is a `useMemo` that only changes if the settings change and the `useEffect` is only concerned with reconnecting the observer if it changes or the ref changes.

We might be overcomplicating this though and having the settings in the effect is okay

Reply:
Good shout, I'll give it a go. Unfortunately, adding the `settings` as a dependency makes the observer always behave as though it's disconnected. I think this might be related to when useMemo() runs, but I'm not certain.

Reply:
I've resolved this issue @marlonmarcello, I think it's much cleaner now. Thanks for the suggestion to memoize the observer. Tested on the local example and it's working perfectly ✨